src/flexalloc_daemon_base.c

```diff
@@ -510,14 +511,30 @@ fla_daemon_fs_init_rq(struct fla_daemon_client *client, int sock_fd)
   client->flexalloc->dev.lb_nbytes = client->flexalloc->geo.lb_nbytes;
   read_ptr += sizeof(struct fla_geo);
 
-  read_ptr += sizeof(size_t); // ignore string length field
-  client->flexalloc->dev.dev_uri = strdup(read_ptr);
+  memcpy(&dev_uri_len, read_ptr, sizeof(size_t));
+  read_ptr += sizeof(size_t);
+  client->flexalloc->dev.dev_uri = strndup(read_ptr, dev_uri_len);
   if (!client->flexalloc->dev.dev_uri)
   {
     err = -ENOMEM;
     return err;
   }
 
+  read_ptr += dev_uri_len;
+  memcpy(&md_dev_uri_len, read_ptr, sizeof(size_t));
+  if(md_dev_uri_len == 0)
+    client->flexalloc->dev.md_dev_uri = NULL;
+  else
+  {
+    read_ptr += sizeof(size_t);
+    client->flexalloc->dev.md_dev_uri = strndup(read_ptr, md_dev_uri_len);
+    if (!client->flexalloc->dev.md_dev_uri)
+    {
+      err = -ENOMEM;
+      return err;
```

Comment:
here you will have to clean up the string allocated @ 516 (`client->flexalloc->dev.dev_uri`)